cmd/openapi.go

```diff
@@ -105,57 +110,99 @@ var openapiDiffCmd = model.ExecutableCommand[OpenAPIDiffFlags]{
 			Required:                   true,
 			AutocompleteFileExtensions: charm_internal.OpenAPIFileExtensions,
 		},
-		flag.EnumFlag{
+		flag.StringFlag{
 			Name:          "output",
 			Shorthand:     "o",
-			Description:   "how to visualize the diff",
+			DefaultValue: "-", // stdout
+			Description:   "output file",
+		},
+		flag.EnumFlag{
+			Name:          "format",
+			Shorthand:     "f",
+			Description:   "output format",
 			AllowedValues: []string{"summary", "console", "html"},
 			DefaultValue:  "summary",
 		},
 	},
 }
 
 func diffOpenapi(ctx context.Context, flags OpenAPIDiffFlags) error {
-	switch flags.Output {
-	case "summary":
-		return changes.RunSummary(flags.OldSchema, flags.NewSchema)
-	case "html":
-		return runHTMLReport(flags, false)
-	case "console":
+	commits, errs := changes.GetChanges(flags.OldSchema, flags.NewSchema, changes.SummaryOptions{});
+	if len(errs) > 0 {
+		return errs[0]
+	}
+	if flags.Format == "summary" {
+		return runSummary(commits)
+	}
+	if flags.Format == "html" {
+		return runHTML(commits, flags, false)
+	}
+	if flags.Format == "console" {
 		return fmt.Errorf("console not supported outside of interactive terminals")
 	}
 
-	return fmt.Errorf("invalid output type: %s", flags.Output)
+	return fmt.Errorf("invalid output type: %s", flags.Format)
 }
 
-func diffOpenapiInteractive(ctx context.Context, flags OpenAPIDiffFlags) error {
-	switch flags.Output {
-	case "summary":
-		return changes.RunSummary(flags.OldSchema, flags.NewSchema)
-	case "html":
-		return runHTMLReport(flags, true)
-	case "console":
-		return runCommand(openapiChanges.GetConsoleCommand(), flags)
+func runHTML(commits []*model2.Commit, flags OpenAPIDiffFlags, shouldOpen bool) error {
+	generator := html_report.NewHTMLReport(false, time.Now(), commits)
+	bytes := generator.GenerateReport(false, false, false)
+	if flags.Output == "-" {
```

Comment:
Not crazy about "-" to mean stdout. Can we just call it "stdout"? or something

Reply:
It's more common place to use `-`. E.g. `grep`, `sed`, `curl`, `awk`, `tar`, `gzip` all establish this convention.